tensorflow/lite/micro/micro_context.h

```diff
@@ -169,12 +171,40 @@ class MicroContext {
     return nullptr;
   }
 
+  struct CustomDecodeRegistration {
+    uint8_t type;       // custom decode type
+    uint8_t reserved1;  // reserved
+    uint8_t reserved2;  // reserved
+    uint8_t reserved3;  // reserved
+    tflite::DecodeState* (*func)(const TfLiteContext*, MicroProfilerInterface*);
```

Comment:
Instead of `func`, `factory` or `create_state` would be better to describes what the function will do.